homework_04/models.py

```diff
@@ -0,0 +1,102 @@
+from sqlalchemy.ext.asyncio import (
+    AsyncEngine,
+    AsyncSession,
+    create_async_engine,
+)
+from sqlalchemy.orm import (
+                declarative_base, 
+                declared_attr,
+                sessionmaker,
+                relationship,
+)
+from sqlalchemy import ( 
+                       Column,
+                       Integer,
+                       String,
+                       Boolean,
+                       ForeignKey,
+                       Text,
+)
+"""
+создайте асинхронный алхимичный engine
+добавьте declarative base (свяжите с engine)
+создайте объект Session на основе класса AsyncSession
+добавьте модели User и Post, объявите поля:
+для модели User обязательными являются name, username, email
+для модели Post обязательными являются user_id, title, body
+создайте связи relationship между моделями: User.posts и Post.user
+"""
+
+import os
+
+PG_URI = "postgresql+pg8000://usr:qwerty@0.0.0.0:5432/homework4"
```

Comment:
при желании, если нужно для alembic, можно сделать так:
https://alembic.sqlalchemy.org/en/latest/cookbook.html

Reply:
"SQLAlchemy version 1.4 introduced experimental support for asyncio, allowing use of most of its interface from async applications. Alembic currently does not provide an async api directly, but it can use an use SQLAlchemy Async engine to run the migrations and autogenerate."

Не смотря на данное высказывание в доке, прикрутить всё получилось. Спасибо !